src/db.rs

```diff
@@ -0,0 +1,57 @@
+use sqlx::{sqlite::SqliteConnectOptions, FromRow, SqlitePool};
+use std::convert::From;
+
+#[derive(FromRow)]
+pub struct Invitation {
+    pub id: i64,
+    pub status: InvitationStatus,
+    pub code: String,
+    pub email: Option<String>,
+}
+
+pub enum InvitationStatus {
+    Pending,
+    Accepted,
+    Rejected,
+}
+
+impl From<String> for InvitationStatus {
+    fn from(status: String) -> Self {
+        match status.as_str() {
+            "Pending" => InvitationStatus::Pending,
+            "Accepted" => InvitationStatus::Accepted,
+            "Rejected" => InvitationStatus::Rejected,
+            _ => panic!("Invalid invitation status"),
+        }
+    }
+}
+
+#[derive(FromRow)]
+pub struct Guest {
+    pub id: i64,
+    pub first_name: String,
+    pub last_name: String,
+    pub invitation_id: i64,
+}
+
+pub async fn create_and_connect() -> SqlitePool {
+    let options = SqliteConnectOptions::new()
+        .create_if_missing(true)
+        .filename("wedding.db");
+    let pool = SqlitePool::connect_with(options).await.unwrap();
+
+    sqlx::migrate!().run(&pool).await.unwrap();
+
+    pool
+}
+
+pub async fn get_invitation_by_code(pool: &SqlitePool, code: &str) -> Option<Invitation> {
+    sqlx::query_as!(
+        Invitation,
+        r#"SELECT id, status, code, email FROM invitations WHERE code = ?"#,
+        code
+    )
+    .fetch_optional(pool)
+    .await
+    .unwrap()
```

Comment:
Don’t unwrap here but let the caller handle the Result as wel as the Option

Reply:
Tried doing that, but then the return type becomes really complex.